presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java

```diff
@@ -496,6 +496,31 @@ public static long stNumGeometries(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
         return ((OGCGeometryCollection) geometry).numGeometries();
     }
 
+    @SqlNullable
+    @Description("Returns a geometry that represents the point set union of the Geometries")
+    @ScalarFunction("ST_Union")
+    @SqlType(GEOMETRY_TYPE_NAME)
+    public static Slice stUnion(@SqlType(GEOMETRY_TYPE_NAME) Slice left, @SqlType(GEOMETRY_TYPE_NAME) Slice right)
+    {
+        // Only supports Geometry but not GeometryCollection due to ESRI library limitation
+        // https://github.com/Esri/geometry-api-java/issues/176
+        // https://github.com/Esri/geometry-api-java/issues/177
+        OGCGeometry leftGeometry = deserialize(left);
+        validateType("ST_Union", leftGeometry, EnumSet.of(POINT, MULTI_POINT, LINE_STRING, MULTI_LINE_STRING, POLYGON, MULTI_POLYGON));
+        OGCGeometry rightGeometry = deserialize(right);
+        validateType("ST_Union", rightGeometry, EnumSet.of(POINT, MULTI_POINT, LINE_STRING, MULTI_LINE_STRING, POLYGON, MULTI_POLYGON));
+
+        if (leftGeometry.isEmpty()) {
```

Comment:
perhaps, move this check above `deserialize(right)` to avoid unnecessary deserialization